ivy/functional/backends/numpy/linear_algebra.py

```diff
@@ -152,10 +152,13 @@ def matrix_norm(
     out: Optional[np.ndarray] = None,
 ) -> np.ndarray:
     if dtype is not None:
-        x = ivy.astype(x, dtype)
+        x = ivy.astype(x, dtype).to_native()
     if not isinstance(axis, tuple):
         axis = tuple(axis)
-    return np.linalg.norm(x, ord=ord, axis=axis, keepdims=keepdims)
+    ret = np.linalg.norm(x, ord=ord, axis=axis, keepdims=keepdims)
+    if ivy.exists(out):
```

Comment:
I think this should get handled automatically through the `handle_out_argument` decorator, could you please check if that's the case? Same applies to the other backends

Reply:
You're 100% correct, I wasn't completely sure about that. Fixed! There are some small changes in torch backend related to the `out` argument as well- with said changes, now all the supported dtypes work correctly (not just float).